src/main/java/io/antmedia/AntMediaApplicationAdapter.java

```diff
@@ -946,69 +999,68 @@ public static String getRelativePath(String filePath){
 	/**
 	 * Notify hook with parameters below
 	 *
-	 * @param url
-	 *            is the url of the service to be called
-	 *
-	 * @param id
-	 *            is the stream id that is unique for each stream
-	 *
-	 * @param action
-	 *            is the name of the action to be notified, it has values such
-	 *            as {@link #HOOK_ACTION_END_LIVE_STREAM}
-	 *            {@link #HOOK_ACTION_START_LIVE_STREAM}
-	 *
-	 * @param streamName,
-	 *            name of the stream. It is not the name of the file. It is just
-	 *            a user friendly name
-	 *
-	 * @param category,
-	 *            category of the stream
-	 *
-	 * @param vodName name of the vod
-	 *
-	 * @param vodId id of the vod in the datastore
-	 *
+	 * @param streamName, name of the stream. It is not the name of the file. It is just
+	 *                    a user friendly name
+	 * @param category,   category of the stream
+	 * @param url         is the url of the service to be called
+	 * @param id          is the stream id that is unique for each stream
+	 * @param mainTrackId mainTrackId(roomId) of the stream
+	 * @param action      is the name of the action to be notified, it has values such
+	 *                    as {@link #HOOK_ACTION_END_LIVE_STREAM}
+	 *                    {@link #HOOK_ACTION_START_LIVE_STREAM}
+	 * @param vodName     name of the vod
+	 * @param vodId       id of the vod in the datastore
 	 * @return
 	 */
-	public void notifyHook(String url, String id, String action, String streamName, String category,
-			String vodName, String vodId, String metadata) {
-		StringBuilder response = null;
-		logger.info("Running notify hook url:{} stream id: {} action:{} vod name:{} vod id:{}", url, id, action, vodName, vodId);
-		if (url != null && url.length() > 0) {
-			Map<String, String> variables = new HashMap<>();
-
-			variables.put("id", id);
-			variables.put("action", action);
+	public void notifyHook(String url, String id, String mainTrackId, String action, String streamName, String category,
+						   String vodName, String vodId, String metadata) {
+		logger.info("Running notify hook url:{} stream id: {} mainTrackId:{} action:{} vod name:{} vod id:{}", url, id, mainTrackId, action, vodName, vodId);
+		if (url != null && !url.isEmpty()) {
+			JSONObject hookPayload = new JSONObject();
```

Comment:
Sending JSON is a good idea. On the other hand, it breaks the backward compatibility. 

What can you do about migrating to JSON gently by keeping backward compatibility?